### DISCUSSION TITLE
TornadoVM v0.15

### DISCUSSION THREAD

**Discussion by jjfumero** [beehive-lab/TornadoVM#227] [announcements]
## Improvements

- New TornadoVM API:
   - API refactoring (``TaskSchedule`` has been renamed to ``TaskGraph``)

   - Introduction of the Immutable ``TaskGraphs``

   - Introduction of the TornadoVM Execution Plans: (``TornadoExecutionPlan``)
   
   - The documentation of migration of existing TornadoVM applications to the new API can be found here: https://tornadovm.readthedocs.io/en/latest/programming.html#migration-to-tornadovm-v0-15 

- Launch a new website https://tornadovm.readthedocs.io/en/latest/ for the documentation
- Improved documentation 
- Initial support for Intel ARC discrete GPUs. 
- Improved TornadoVM installer for Linux 
- ImprovedTornadoVM launch script with optional parameters
- Support of large buffer allocations with Intel Level Zero. Use: ``tornado.spirv.levelzero.extended.memory=True``

## Bug fixes:

- Vector and Matrix types 
- TornadoVM Floating Replacement compiler phase fixed
- Fix ``CMAKE`` for Intel ARC GPUs
- Device query tool fixed for the PTX backend
- Documentation for Windows 11 fixed

## POM file dependencies
To obtain the TornadoVM API from maven dependencies, you need to include the following dependency in your pom file. Note that, for running the TornadoVM application, you either need: a) TornadoVM SDK Binaries; b) a full installation of the TornadoVM; or c) a docker instance of TornadoVM. 

```xml
<repositories>
    <repository>
        <id>universityOfManchester-graal</id>
        <url>https://raw.githubusercontent.com/beehive-lab/tornado/maven-tornadovm</url>
    </repository>
</repositories>

<dependencies>
<dependency>
    <groupId>tornado</groupId>
    <artifactId>tornado-api</artifactId>
    <version>0.15</version>
</dependency>
<dependency>
    <groupId>tornado</groupId>
    <artifactId>tornado-matrices</artifactId>
    <version>0.15</version>
</dependency>
</dependencies>

<hr /><em>This discussion was created from the release <a href='https://github.com/beehive-lab/TornadoVM/releases/tag/v0.15'>TornadoVM v0.15</a>.</em>

**Comment by zealbell** [comment]
Hi, happy new year. Are there sources and javadocs available for tornado-api-0.15.jar or tornado-api-0.16-dev.jar. In my maven project I have tried:

```bash
$ mvn dependency:sources -Dsilent=true
$ mvn dependency:resolve -Dclassifier=javadoc
```
still nothing, everything else gets downloaded with the exception of tornado

**Reply by jjfumero** [reply]
We haven't published the java docs yet. For reference of using the TornadoVM APIs, we recommend:

- The Documentation website: https://tornadovm.readthedocs.io/en/latest/programming.html#
- The examples suite from TornadoVM: https://github.com/beehive-lab/TornadoVM/tree/master/tornado-examples/src/main/java/uk/ac/manchester/tornado/examples
